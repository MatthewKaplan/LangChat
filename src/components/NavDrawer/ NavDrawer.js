import React, { Component } from 'react';
import { Text, View, TouchableOpacity, Image, SafeAreaView } from 'react-native';
import Drawer from 'react-native-drawer';
import { Ionicons } from '@expo/vector-icons';
import Nav from '../Nav/Nav';
console.disableYellowBox = true;

export default class NavDrawer extends Component {
	constructor (props) {
		super(props);
	}

	renderDrawer () {
		return (
			<View style={styles.menuContainer}>
				<Nav />
			</View>
		);
	}

	openDrawer = () => {
		this.drawer.open();
	};

	closeDrawer = () => {
		this.drawer.close();
	};

	render () {
		return (
			<SafeAreaView style={styles.safeAreaStyle}>
				<View style={styles.mainContainer}>
					<Drawer
						ref={ref => (this.drawer = ref)}
						content={this.renderDrawer()}
						type="static"
						tapToClose={true}
						openDrawerOffset={0.25}
						styles={drawerStyles}>
						<View style={styles.headerContainer}>
							<TouchableOpacity style={styles.menuButton} onPress={this.openDrawer}>
								<Ionicons name="ios-menu" style={styles.menuIcon} size={40} onPress={this.openDrawer} />
							</TouchableOpacity>
							<Text style={styles.headerTitle}>LangChat</Text>
							<View style={styles.menuButton} />
						</View>
						{this.props.children || null}
					</Drawer>
				</View>
			</SafeAreaView>
		);
	}
}

const drawerStyles = {
	drawer: {
		flex: 1.0,

		backgroundColor: '#3B5998'
	},
	main: {
		flex: 1.0,
		backgroundColor: 'white'
	}
};

const styles = {
	mainContainer: {
		flex: 1.0,
		backgroundColor: 'white'
	},
	safeAreaStyle: {
		flex: 1.0,
		backgroundColor: '#3B5998'
	},
	headerContainer: {
		position: 'relative',
		alignItems: 'center',
		height: 44,
		flexDirection: 'row',
		justifyContent: 'center',
		backgroundColor: '#3B5998'
	},
	headerTitle: {
		flex: 1.0,
		textAlign: 'center',
		alignSelf: 'center',
		color: 'white'
	},
	menuButton: {
		marginLeft: 8,
		marginRight: 8,
		alignSelf: 'center',
		tintColor: 'white'
	},
	menuContainer: {
		flex: 1.0,
		backgroundColor: '#3B5998'
	},
	menuTitleContainer: {
		height: 60,
		width: '100%',
		flexDirection: 'row',
		alignItem: 'center'
	},
	menuTitle: {
		width: '100%',
		color: 'white',
		fontSize: 17,
		alignSelf: 'center',
		textAlign: 'center'
	},
	menuIcon: {
		color: 'white'
	}
};
