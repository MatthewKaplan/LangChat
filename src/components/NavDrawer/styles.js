import { StyleSheet } from 'react-native';

export default StyleSheet.create({
	mainContainer: {
		flex: 1.0,
		backgroundColor: '#fff'
	},
	safeAreaStyle: {
		flex: 1.0,
		backgroundColor: '#fff'
	},
	headerContainer: {
		position: 'relative',
		height: 45,
		flexDirection: 'row',
		justifyContent: 'center',
		borderBottomColor: '#a7a7a7',
		borderBottomWidth: 1
	},
	headerTitle: {
		flex: 1.0,
		position: 'absolute',
		textAlign: 'center',
		alignSelf: 'center',
		color: '#323232',
		fontSize: 22
	},
	menuButton: {
		alignSelf: 'center',
		tintColor: 'white',
		marginLeft: 5
	},
	menuContainer: {
		flex: 1.0,
		backgroundColor: '#323232'
	},
	menuTitleContainer: {
		height: 60,
		width: '100%',
		flexDirection: 'row',
		alignItems: 'center',
		position: 'absolute'
	},
	menuTitle: {
		width: '100%',
		color: 'white',
		fontSize: 17,
		alignSelf: 'center',
		textAlign: 'center',
		marginRight: '2%'
	},
	menuIcon: {
		color: '#323232',
		zIndex: 9999,
		marginTop: 2,
		marginLeft: 5,
		alignSelf: 'flex-start',
		justifyContent: 'center',
		flex: 1.0
	}
});

export const drawerStyles = {
	drawer: {
		flex: 1.0
	},
	main: {
		flex: 1.0,
		backgroundColor: 'white'
	}
};
