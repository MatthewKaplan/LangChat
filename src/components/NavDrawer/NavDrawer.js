import React, { Component } from 'react';
import { Text, View, TouchableOpacity, SafeAreaView, StatusBar } from 'react-native';
import Drawer from 'react-native-drawer';
import { AntDesign } from '@expo/vector-icons';
import { connect } from 'react-redux';
import { Actions } from 'react-native-router-flux';
import { logout, handleError } from '../../actions';
import PropTypes from 'prop-types';
import Nav from '../Nav/Nav';
import { drawerStyles, styles } from './styles';

console.disableYellowBox = true;

export class NavDrawer extends Component {
	state = {
		drawerOpen: false
	};

	renderDrawer () {
		return (
			<View style={styles.menuContainer}>
				<Nav />
			</View>
		);
	}

	openDrawer = () => {
		this.drawer.open();
	};

	closeDrawer = () => {
		this.drawer.close();
	};

	toggleDrawer = () => {
		const { drawerOpen } = this.state;
		this.setState({ drawerOpen: !drawerOpen });
	};

	logout = async () => {
		try {
			const { username, password } = this.props;
			await fetch('https://langchat-crosspollination.herokuapp.com/api/v1/log_out/', {
				method: 'POST',
				headers: {
					'Content-Type': 'application/json',
					'X-CSRFToken': this.props.cookie,
					Authorization: `Token ${this.props.token}`
				},
				body: JSON.stringify({ username, password })
			});
			this.props.logout();
			Actions.splashPage();
		} catch (error) {
			this.props.handleError(error.message)
		}
	};

	render () {
		return (
			<SafeAreaView style={styles.safeAreaStyle}>
				<StatusBar barStyle="dark-content" />
				<View style={styles.mainContainer}>
					<Drawer
						ref={ref => (this.drawer = ref)}
						content={this.renderDrawer()}
						type="static"
						onOpen={this.toggleDrawer}
						onClose={this.toggleDrawer}
						tapToClose={true}
						openDrawerOffset={0.25}
						styles={drawerStyles}>
						<View style={styles.headerContainer}>
							<TouchableOpacity style={styles.menuButton} onPress={this.openDrawer} style={styles.menuIcon}>
								{this.state.drawerOpen ? (
									<AntDesign
										name="menu-unfold"
										style={styles.menuIcon}
										size={25}
										onPress={() => this.openDrawer()}
										data-test="close-drawer-btn"
									/>
								) : (
									<AntDesign
										name="menu-fold"
										style={styles.menuIcon}
										size={25}
										onPress={() => this.openDrawer()}
										data-test="open-drawer-btn"
									/>
								)}
							</TouchableOpacity>
							<Text style={styles.headerTitle}>{this.props.name || 'LangChat'}</Text>
							<View style={styles.menuButton} />
						</View>
						{this.props.children || null}
					</Drawer>
				</View>
			</SafeAreaView>
		);
	}
}

NavDrawer.propTypes = {
	logout: PropTypes.func.isRequired,
	cookie: PropTypes.string.isRequired,
	username: PropTypes.string.isRequired,
	password: PropTypes.string.isRequired,
	token: PropTypes.string.isRequired
};

export const mapStateToProps = state => ({
	cookie: state.cookie,
	username: state.auth.username,
	password: state.auth.password,
	token: state.token
});

export const mapDispatchToProps = dispatch => ({
	logout: () => dispatch(logout()),
	handleError: errorMessage => dispatch(handleError(errorMessage))
});

export default connect(mapStateToProps, mapDispatchToProps)(NavDrawer);
