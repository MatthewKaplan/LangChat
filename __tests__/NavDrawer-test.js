// __tests__/NavDrawer-test.js
import React from 'react';
import { NavDrawer } from '../src/components/NavDrawer/NavDrawer';

import renderer from 'react-test-renderer';

describe('NavDrawer Component', () => {
	it('should match snapshot', () => {
		const tree = renderer.create(<NavDrawer />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
