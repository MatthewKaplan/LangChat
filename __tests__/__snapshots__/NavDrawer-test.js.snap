// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDrawer Component should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#3B5998",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3B5998",
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        elevation={0}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#3B5998",
              "borderWidth": 0,
              "flex": 1,
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
            Object {
              "height": 1334,
              "width": 562.5,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#3B5998",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "color": "white",
                "flex": 1,
                "marginTop": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "marginTop": 10,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
              >
                <Text />
              </View>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                style={
                  Object {
                    "color": "#E8EAEF",
                    "fontSize": 20,
                    "padding": 10,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                splash
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                Text={true}
                style={
                  Object {
                    "color": "#E8EAEF",
                    "fontSize": 20,
                    "padding": 10,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                login
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                Text={true}
                style={
                  Object {
                    "color": "#E8EAEF",
                    "fontSize": 20,
                    "padding": 10,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                signup
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                Text={true}
                style={
                  Object {
                    "color": "#E8EAEF",
                    "fontSize": 20,
                    "padding": 10,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                home
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                Text={true}
                style={
                  Object {
                    "color": "#E8EAEF",
                    "fontSize": 20,
                    "padding": 10,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                chatroom french
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                Text={true}
                style={
                  Object {
                    "color": "#E8EAEF",
                    "fontSize": 20,
                    "padding": 10,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                chatroom spanish
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                Text={true}
                style={
                  Object {
                    "color": "#E8EAEF",
                    "fontSize": 20,
                    "padding": 10,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                tutorial
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                Text={true}
                style={
                  Object {
                    "color": "#E8EAEF",
                    "fontSize": 20,
                    "padding": 10,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                tutorial
              </Text>
            </View>
          </View>
        </View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderWidth": 0,
              "flex": 1,
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
            Object {
              "height": 1334,
              "width": 750,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3B5998",
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "marginLeft": 8,
                "marginRight": 8,
                "opacity": 1,
                "tintColor": "white",
              }
            }
          >
            <Text />
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
                "flex": 1,
                "textAlign": "center",
              }
            }
          >
            LangChat
          </Text>
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginLeft": 8,
                "marginRight": 8,
                "tintColor": "white",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
